src/widgets/barchart/bar.rs

```diff
@@ -43,14 +48,33 @@ pub struct Bar<'a> {
 }
 
 impl<'a> Bar<'a> {
+    /// Creates a new `Bar` with the given label and value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ratatui::widgets::Bar;
+    ///
+    /// let bar = Bar::new("Label", 42);
+    /// ```
+    pub fn new(label: impl Into<Line<'a>>, value: u64) -> Self {
```

Comment:
Also, what do you think of renaming this to `with_label` which I mentioned above?

Reply:
I agree with both of your opinions. I think we should modify the `new` as it will only take the value and also have another impl `with_label` for more specific usages.